packages/ses/src/module-instance.js

```diff
@@ -0,0 +1,335 @@
+            get,
+            set,
+            enumerable: true,
+            configurable: false,
+          });
+        }
+        liveVar[localName] = update;
+      }
+
+      moduleProps[liveExportName] = {
+        get: liveGetNotify.get,
+        set: undefined,
+        enumerable: true,
+        configurable: false,
+      };
+
+      notifiers[liveExportName] = liveGetNotify.notify;
+    },
+  );
+
+  const notifyStar = update => {
+    update(module);
+  };
+  notifiers['*'] = notifyStar;
+
+  // The updateRecord must conform to moduleAnalysis.imports
+  // updateRecord = Map<specifier, importUpdaters>
+  // importUpdaters = Map<importName, [update(newValue)*]>
+  function imports(updateRecord) {
+    // By the time imports is called, the importedInstances should already be
+    // initialized with module instances that satisfy
+    // imports.
+    // importedInstances = Map[_specifier_, { notifiers, module, execute }]
+    // notifiers = { _importName_: notify(update(newValue))}
+
+    // export * cannot export default.
+    const candidateAll = create(null);
+    candidateAll.default = false;
+    for (const [specifier, importUpdaters] of updateRecord.entries()) {
+      const instance = importedInstances.get(specifier);
+      instance.execute(); // bottom up cycle tolerant
+      const { notifiers: modNotifiers } = instance;
+      for (const [importName, updaters] of importUpdaters.entries()) {
+        const notify = modNotifiers[importName];
+        if (!notify) {
+          throw SyntaxError(
+            `The requested module '${specifier}' does not provide an export named '${importName}'`,
+          );
+        }
+        for (const updater of updaters) {
+          notify(updater);
+        }
+      }
+      if (exportAlls.includes(specifier)) {
+        // Make all these imports candidates.
+        for (const [importName, notify] of entries(modNotifiers)) {
+          if (candidateAll[importName] === undefined) {
+            candidateAll[importName] = notify;
+          } else {
+            // Already a candidate: remove ambiguity.
+            candidateAll[importName] = false;
+          }
+        }
+      }
+    }
+
+    for (const [importName, notify] of entries(candidateAll)) {
+      if (!notifiers[importName] && notify !== false) {
+        notifiers[importName] = notify;
+
+        // exported live binding state
+        let value;
+        notify(v => (value = v));
+        moduleProps[importName] = {
+          get() {
+            return value;
+          },
+          set: undefined,
+          enumerable: true,
+          configurable: false,
+        };
+      }
+    }
+
+    // Sort the module exports namespace as per spec.
+    // The module exports namespace will be wrapped in a module namespace
+    // exports proxy which will serve as a "module exports namespace exotic
+    // object".
+    keys(moduleProps)
+      .sort()
+      .forEach(k => defProp(module, k, moduleProps[k]));
```

Comment:
I believe this satisfies the ordering requirement. We could also enforce the ordering requirement later on the module namespace proxy by trapping ownKeys. We could also use a second class for loop here since we don’t need the closure. I’m inclined to run past this since in many other places in this module, the closure is necessary.